modules.md

```diff
@@ -0,0 +1,233 @@
+  - Files in a directory are always in scope as modules in the corresponding file. E.g., if `bar.kcl` exists in the directory `foo`, then `bar` is available as a module name inside `foo.kcl`.
+
+### Paths
+
+A module variable is always required to start a path. Modules are always considered exported. Starting a path with `::` will refer to any items in the root of the project.
+
+```
+import "foo.kcl"
+foo::bar // ok
+foo::bar::qux // ok
+baz::bar // ok if bar exists inside baz.kcl in a subdirectory with the same name as the current file
+```
+
+### Namespacing
+
+Technical note:
+
+Modules and non-module items are in separate namespaces (that is, modules and non-module items cannot be used in the same syntactic space, except when imported). So local names and sub-modules may exist with the same name. This also makes sense when considering files as parts - the file is both a part and a module, but they are not the same object. When importing a name, it imports the name in all namespaces in which it is exported.
+
+Note that in the future, types will also be in a separate namespace.
+
+
+## Importing from other directories within the project
+
+File paths (as well as filenames) may be used inside the quoted string of an `import`. Paths may contain `/` (but not `\`, even on Windows). Paths are always relative to the top-level of the project (not relative to the current file or absolute paths), and may not include `.` or `..` (alternative: we could relax this requirement and allow relative paths as long as traversing the path never leaves the project directory) or global identifiers (`c:`, etc. or starting with '/'). Paths are ignored in the name of the module as imported. E.g., `import "foo.kcl"` brings in `foo` from the root directory of the project, `import "baz/bar/foo.kcl"` brings in `foo` from `foo.kcl` found at `baz/bar/`, `import a from "baz/bar/foo.kcl"` brings in `a` from the same file. The path of the file makes no difference to the imported name or to privacy.
+
+Directories cannot be imported or imported from, e.g., `import foo from "baz/bar"` is not allowed.
+
+All modules within a project are nameable, there is no need for `export` anywhere on a module. If all items in a module are private (i.e., none are `export`ed), the module itself can still be named, but it is useless (like an empty object) and a glob import would import nothing.
+
+## Imports from the standard library
+
+Imports from the standard library work like other imports, but the import path begins with `std` and has no `.kcl` extension. The interpreter knows where to find the standard library which works like any other KCL project.
+
+There is an implicit import `import * from 'std'` in every file unless it includes a `@no_std` attribute. Items in `std` can be imported too. There is no explicit prelude, all of std is treated like a prelude.
```

Comment:
To clarify my understanding, if there's a math module in std with `cos()`, `import * from 'std'` imports everything in std (one level deep), so the module math? So users need to say `math::cos(x)`?

Reply:
Yes